Phase out 'Display' Contexts: Surface some Conditions in Block Visibility

JojoVes opened the pull request:
**Drupal Issue**: https://www.drupal.org/project/islandora/issues/3536145

* Other Relevant Links:
  * https://docs.google.com/spreadsheets/d/1WTcsRPm178dYrurpW1Cqs_MyUc-w1I1s68mq403xs6g/edit?gid=0#gid=0
  * https://www.drupal.org/project/islandora/issues/3536145#comment-16194229

# What does this Pull Request do?

Stop unsetting the 'node_has_term' and 'node_is_islandora_object' conditions in Block Visibility form to allow moving away from relying on Contexts to place various blocks.

# What's new?
This should not affect any existing configuration; it just facilitates changing block placements currently relying on various Contexts to instead use Block Visibility configuration directly.

# How should this be tested?

* after pulling in this change, the Block Visibility section of block configurations should include 'Node has term with URI' and 'Node is an Islandora node' among the available tabs.
  * try setting either one or both or none of these visibility conditions for any block.
  * these conditions should work however they are configured, and should not have any required fields to cause issues when not used (which had been the motivation for hiding at least a few of those other conditions being unset)

# Additional Notes
I did also include a small bit of cleanup so the MediaSourceHasMimetype condition has the proper default config defined so it can stop getting thrown into all the block configs that are not using it.

# Documentation Status

Don't think this should require documentation changes, though might be nice to put a blurb somewhere to suggest avoiding using Contexts to place blocks in favour of Block Visibility?

# Interested parties
@Islandora/committers

JojoVes committed:
let 'node_has_term' and 'node_is_islandora_object' be in block visibility conditions

JojoVes committed:
add config field to NodeIsIslandoraObject condition

Conditions only get applied in Block Visibility when there's a config
change for that condition in the block config form, so it needs a field
to configure.

JojoVes committed:
more informative field description

adam-vessey reviewed (commented):
I think moving
```
 *     "node" = @ContextDefinition("entity:node", required = TRUE , label = @Translation("node"))
 ```
to 
```
 *     "node" = @ContextDefinition("entity:node", required = FALSE , label = @Translation("node"))
 ```
might be the more straight-forward approach?

That said, the behaviour when not having the context value is always a little weird, conceptually?

JojoVes committed:
some cleanup to config for NodeIsIslandoraObject

Much thanks to @adam-vessey for helping to figure this out!

JojoVes committed:
cleanup default config for MediaSourceHasMimetype

Might as well get this cleaned up so it stops inserting itself in all
the block configs where it's not relevant.

JojoVes committed:
remove unused use statement

rosiel reviewed (approved):
Somehow the node source being optional makes this whole condition optional, which is great. Further, it makes it work and negation works. Tested, i approve.
